### PR TITLE
Move search result processing to core

### PR DESCRIPTION
**Pull request by mathiascode**
This hasn't been extensively tested nor optimized yet.

### CODE REVIEW

**Comment by mathiascode** [comment]
Once this is merged, we can start moving filters to the core and port the treeview to our TreeView class.

**Comment by mathiascode** [comment]
Marking this as ready for testing. Some of the performance concerns when filtering will be addressed in the TreeView port.

File: pynicotine/gtkgui/search.py [hunk 1]
```diff
@@ -1397,21 +1311,22 @@ def on_download_folders(self, *_args, download_location=""):
...
                 destination = core.transfers.get_folder_destination(user, folder, remove_destination=False)
 
-                (_counter, user, _flag, _h_speed, _h_queue, _directory, _file_type, _filename,
-                    _h_size, h_bitrate, h_length, _bitrate, fullpath, _country_code, size, _speed,
-                    _queue, _length, _has_free_slots) = row
-                visible_files.append(
-                    (user, fullpath, destination, size.get_value(), h_bitrate, h_length))
+                size = result.size
+                h_bitrate, _bitrate, h_length, _length = slskmessages.FileListMessage.parse_result_bitrate_length(
+                    size, result.attributes)
```
Comment (slook) [inline pynicotine/gtkgui/search.py:1327]: This doesn't seem to belong here.

Consider processing `result.attributes` and `destination` in `core.search.request_folder_download()` instead...

`visible_files.append((user, file_path, result.size, result.attributes))`

File: pynicotine/search.py [hunk 2]
```diff
@@ -47,6 +48,44 @@ def __init__(self, token=None, term=None, mode="global", room=None, users=None,
...
+
+    __slots__ = ("num", "user", "file_path", "size", "attributes", "speed", "country_code", "queue",
+                 "has_free_slots", "is_private")
+
+    def __init__(self, num=None, user=None, file_path=None, size=None, attributes=None, speed=None, country_code=None,
+                 queue=None, has_free_slots=None, is_private=False):
+
+        self.num = num
+        self.user = user
+        self.file_path = file_path
```
Comment (slook) [inline pynicotine/search.py:80]: `file_path` -> `fullpath` or `full_path`

Also consider adding `folder` and `filename` here would help to reduce the amount of split() and join() operations elsewhere.

File: pynicotine/search.py [hunk 3]
```diff
@@ -289,6 +340,50 @@ def do_wishlist_search_interval(self):
...
+                    "query": search.term
+                })
+                continue
+
+            search.num_results += 1
+            search.results[user].append(
+                SearchResult(
+                    num=search.num_results,
+                    user=user,
+                    file_path=fullpath,
```
Comment (slook) [inline pynicotine/search.py:376]: `file_path` -> `fullpath` (or unify all to `full_path`)

Also consider adding `folder`** and `filename` into the `SearchResult()` would help to reduce the amount of rsplit() and join() operations elsewhere.

** _Maybe `dir` or `directory` is appropriate for references to remote folders._

File: pynicotine/search.py [hunk 4]
```diff
@@ -47,6 +48,44 @@ def __init__(self, token=None, term=None, mode="global", room=None, users=None,
...
+
+class SearchResult:
+
+    __slots__ = ("num", "user", "file_path", "size", "attributes", "speed", "country_code", "queue",
+                 "has_free_slots", "is_private")
+
+    def __init__(self, num=None, user=None, file_path=None, size=None, attributes=None, speed=None, country_code=None,
+                 queue=None, has_free_slots=None, is_private=False):
+
+        self.num = num
```
Comment (slook) [inline pynicotine/search.py:78]: `num` -> `id` or `result_id`

File: pynicotine/search.py (hunk 3 above)
Comment (slook) [reply pynicotine/search.py:376]: The `path` term is too ambiguous as to a folder without a filename.

Consider adding `folder`** and `filename` into the `SearchResult()` would help to reduce the amount of rsplit() and join() operations elsewhere.

** or maybe `dir` or `directory` would be more appropriate for references to remotely indexed folders.

Comment (slook) [reply pynicotine/search.py:376]: Disregard this review, I agree that file_path makes the most sense.

**Comment by slook** [comment]
Surely this PR is way too old and is irrelevant now, or is it not to complex to be rebased?

**Comment by mathiascode** [comment]
Some parts are relevant, but I don't have time to work on this now.

**Pull request closed without merging**